### PR TITLE
added ZAP configuration for proper time format

### PR DESCRIPTION
**Pull request by bisztray**

### CODE REVIEW

File: consul-operator/main.go [hunk 1]
```diff
@@ -64,7 +83,7 @@ func main() {
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts), zap.Encoder(zapcore.NewConsoleEncoder(logEncoderConfig))))
```
Comment (howardhalili) [inline consul-operator/main.go:86]: maybe it's more simple to just add `TimeEncoder: zapcore.ISO8601TimeEncoder` in opts variable as part of zap.Options.

Comment (bisztray) [reply consul-operator/main.go:86]: If you only add the Iso8601TimeEncoder, then by default the caller is not logged only the timestamp and the message. This way the caller is logged as well.

Comment (howardhalili) [reply consul-operator/main.go:86]: ok, so you also want the caller. with `Development: true`, stacktrace is already available from warning level.
anyway, log content is usually subjective & depends on preference. not a problem.

**Review by howardhalili — APPROVED** [review]

File: consul-operator/componenttest/suite_initializer_test.go [hunk 2]
```diff
@@ -16,18 +16,42 @@ import (
...
+var encoderConfig = zapcore.EncoderConfig{
+	MessageKey: "message",
+
+	LevelKey:    "level",
+	EncodeLevel: zapcore.CapitalLevelEncoder,
+
+	FunctionKey: "function",
+
+	TimeKey:    "time",
+	EncodeTime: zapcore.ISO8601TimeEncoder,
```
Comment (hochdorf) [inline consul-operator/componenttest/suite_initializer_test.go:42]: Pls logs UTC time.
You can achieve it with a custom time encoder:
```
func CustomUTCISO8601TimeEncoder(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
	layout := "2006-01-02T15:04:05.000Z0700"
	type appendTimeEncoder interface {
		AppendTimeLayout(time.Time, string)
	}

	if enc, ok := enc.(appendTimeEncoder); ok {
		enc.AppendTimeLayout(t.UTC(), layout)
		return
	}

	enc.AppendString(t.UTC().Format(layout))
}
```

Comment (bisztray) [reply consul-operator/componenttest/suite_initializer_test.go]: done

**Review by hochdorf — APPROVED** [review]

**Pull request merged**